### PR TITLE
update misc/hill-climbing

### PR DESCRIPTION
**Pull request by isdanni**
**审核的同学** 请着重关注以下四方面：

1. 注意有没有 typo
2. 不论你是否熟悉相关知识，都请以初学者的角度把这个 PR 的内容阅读一遍，跟着作者的思路走，然后谈谈你的感受
3. 如果你熟悉相关知识，请按照自己的理解评估这个 PR 的内容是否合适
4. 请**尽量**保持跟进直到它被 merge 或 close

### CODE REVIEW

File: docs/misc/hill-climbing.md [hunk 1]
```diff
@@ -1,49 +1,50 @@
...
+Therefore, the hill climbing algorithm searches for a new solution near the currently found optimal solution $x$ each time. If this new $x'$ is better, then transfer to $x'$ , otherwise it remains unchanged.
 
-这种算法对于单峰函数显然可行。
+This algorithm is obviously feasible for unimodal functions.
 
-> Q：你都知道是单峰函数了为什么不三分呢
-> A：在多年的 OI 生活中，我意识到了，人类是有极限的，无论多么工于心计，绞尽脑汁，状态总是表示不出来的，出题人的想法总是猜不透的，边界总是写不对的——所以——我不三分了 JOJO！
+> Joke time:
+> Q: Now that it is a unimodal function, why not use ternary search?
+> A：In many years of OI, I have realized that human beings have their limits, no matter how scheming they are. The state is always unrepresentable. We can never get the mind of the people who create the problem and the boundary written is always wrong -- so -- I won't use ternary search anymore JOJO!
```
Comment (ksyx) [inline docs/misc/hill-climbing.md:13]: JOJO 可以删了？:new_moon_with_face:
另外单峰函数的翻译有没有 ref 鸭

File: docs/misc/hill-climbing.md [hunk 2]
```diff
@@ -1,49 +1,50 @@
...
+## Introduction
 
-爬山算法是一种局部择优的方法，采用启发式方法，是对深度优先搜索的一种改进，它利用反馈信息帮助生成解的决策。
+The hill climbing algorithm is a method of local selection. It uses a heuristic method and is an improvement to DFS. It uses feedback information to help generate solution decisions.
 
-直白地讲，就是当目前无法直接到达最优解，但是可以判断两个解哪个更优的时候，根据一些反馈信息生成一个新的可能解。
+Frankly speaking, when the optimal solution cannot be reached directly, but it is possible to determine which of the two solutions is better, a new possible solution is generated based on some feedback information.
 
-因此，爬山算法每次在当前找到的最优方案 $x$ 附近寻找一个新方案。如果这个新的解 $x'$ 更优，那么转移到 $x'$ ，否则不变。
+Therefore, the hill climbing algorithm searches for a new solution near the currently found optimal solution $x$ each time. If this new $x'$ is better, then transfer to $x'$ , otherwise it remains unchanged.
```
Comment (ksyx) [inline docs/misc/hill-climbing.md:7]: move to

File: docs/misc/hill-climbing.md [hunk 3]
```diff
@@ -1,49 +1,50 @@
...
+This algorithm is obviously feasible for unimodal functions.
 
-> Q：你都知道是单峰函数了为什么不三分呢
-> A：在多年的 OI 生活中，我意识到了，人类是有极限的，无论多么工于心计，绞尽脑汁，状态总是表示不出来的，出题人的想法总是猜不透的，边界总是写不对的——所以——我不三分了 JOJO！
+> Joke time:
+> Q: Now that it is a unimodal function, why not use ternary search?
+> A：In many years of OI, I have realized that human beings have their limits, no matter how scheming they are. The state is always unrepresentable. We can never get the mind of the people who create the problem and the boundary written is always wrong -- so -- I won't use ternary search anymore JOJO!
 
-认真地说，爬山算法的优势在于当正解的写法你并不了解（常见于毒瘤计算几何和毒瘤数学题），或者本身状态维度很多，无法容易地写分治（例 2 就可以用二分完成合法正解）时，可以通过非常暴力的计算得到最优解。
+Seriously speaking, the advantage of the hill climbing algorithm is that when you don’t know how to write formal solutions (commonly used in geometry calculation and mathematics problems), or there are many dimensions of its own state, and it is not easy to write divide and conquer (sample 2 can be completed by binary search), the optimal solution can be obtained through brute force.
```
Comment (ksyx) [inline docs/misc/hill-climbing.md:15]: many dimensions when representing a state
```suggestion
Seriously speaking, the advantage of the hill climbing algorithm is that when you don’t know how to write formal solutions (commonly used in geometry calculation and mathematics problems), or when there are many dimensions of its own state, making it hard to implement divide and conquer (sample 2 can be completed by binary search), the optimal solution can be obtained through brute force.
```

File: docs/misc/hill-climbing.md [hunk 4]
```diff
@@ -1,49 +1,50 @@
...
+The process of the rabbit gradually becoming awake is the cooling process, that is, the temperature parameter will continue to decrease when climbing the mountain.
 
-关于降温：降温参数是略小于 $1$ 的常数，一般在 $[0.985, 0.999]$ 中选取。
+About cooling: The cooling parameter is a constant slightly less than $1$ , generally selected within $[0.985, 0.999]$ .
 
-### 例 1 [「JSOI2008」球形空间产生器](https://www.luogu.com.cn/problem/P4035) 
+### Sample 1 [「JSOI2008」Spherical space generator](https://www.luogu.com.cn/problem/P4035) 
 
-题意：给出 $n$ 维空间中的 $n$ 个点，已知它们在同一个 $n$ 维球面上，求出球心。 $n \leq 10$ ，坐标绝对值不超过 $10000$ 。
+Problem: Given $n$ points in $n$-dimensional space, knowing that they are on the same $n$-dimensional sphere, find the center of the sphere. $n \leq 10$ , and the absolute value of the coordinate does not exceed $10000$ .
```
Comment (ksyx) [inline docs/misc/hill-climbing.md:33]: coordinates

File: docs/misc/hill-climbing.md [hunk 5]
```diff
@@ -1,49 +1,50 @@
...
 
-很明显的单峰函数，可以使用爬山解决。本题算法流程：
+Obviously, the unimodal function can be solved by hill climbing. The algorithm of this question:
```
Comment (ksyx) [inline docs/misc/hill-climbing.md:35]:
```suggestion
Obviously, the unimodal function can be solved by hill climbing. The algorithm of this question is as follows:
```

File: docs/misc/hill-climbing.md [hunk 6]
```diff
@@ -1,49 +1,50 @@
...
 
-1. 初始化球心为各个给定点的重心（即其各维坐标均为所有给定点对应维度坐标的平均值），以减少枚举量。
-2. 对于当前的球心，求出每个已知点到这个球心欧氏距离的平均值。
-3. 遍历所有已知点。记录一个改变值 cans（分开每一维度记录）对于每一个点的欧氏距离，如果大于平均值，就把改变值加上差值，否则减去。实际上并不用判断这个大小问题，只要不考虑绝对值，直接用坐标计算即可。这个过程可以形象地转化成一个新的球心，在空间里推来推去，碰到太远的点就往点的方向拉一点，碰到太近的点就往点的反方向推一点。
-4. 将我们记录的 cans 乘上温度，更新球心，回到步骤 2
-5. 在温度小于某个给定阈值的时候结束。
+1. Initialize the center of the sphere to the center of gravity of each given point (that is, its dimensional coordinates are the average value of the dimensional coordinates of all the given points) to reduce enumeration operations.
+2. For the current sphere center , find the average of the Euclidean distance from each known point to the center of the sphere.
```
Comment (ksyx) [inline docs/misc/hill-climbing.md:38]:
```suggestion
2. For the current sphere center, find the average of the Euclidean distance from each known point to the center of the sphere.
```

File: docs/misc/hill-climbing.md [hunk 7]
```diff
@@ -1,49 +1,50 @@
...
+3. Traverse all known points. Record a change value cans (record each dimension separately) for the Euclidean distance of each point. If it is greater than the average value, add the difference to the change value, otherwise subtract it. In fact, there is no need to check the size, as long as the absolute value is not considered, the coordinates can be directly calculated. This process can be transformed into a new center of the sphere pushing it back and forth, pulling it in the direction of the point when it touches a point too far away, and pushing it in the opposite direction when it touches a point too close.
+4. Multiply the recorded cans by the temperature, update the center of the sphere, and go back to step 2.
```
Comment (ksyx) [inline docs/misc/hill-climbing.md:40]: 这里 cans 用 $\mathit{}$ 包起来吧

File: docs/misc/hill-climbing.md [hunk 8]
```diff
@@ -1,49 +1,50 @@
...
-4. 将我们记录的 cans 乘上温度，更新球心，回到步骤 2
-5. 在温度小于某个给定阈值的时候结束。
+1. Initialize the center of the sphere to the center of gravity of each given point (that is, its dimensional coordinates are the average value of the dimensional coordinates of all the given points) to reduce enumeration operations.
+2. For the current sphere center , find the average of the Euclidean distance from each known point to the center of the sphere.
+3. Traverse all known points. Record a change value cans (record each dimension separately) for the Euclidean distance of each point. If it is greater than the average value, add the difference to the change value, otherwise subtract it. In fact, there is no need to check the size, as long as the absolute value is not considered, the coordinates can be directly calculated. This process can be transformed into a new center of the sphere pushing it back and forth, pulling it in the direction of the point when it touches a point too far away, and pushing it in the opposite direction when it touches a point too close.
```
Comment (ksyx) [inline docs/misc/hill-climbing.md:39]: > pulling it in the direction of the point when it touches a point too far away, and pushing it in the opposite direction when it touches a point too close.

{pull|push}ing it when it is too {far|close} to the center of sphere.

File: docs/misc/hill-climbing.md [hunk 9]
```diff
@@ -73,24 +74,26 @@
...
+        for (int i = 1; i <= n; i++) ans[i] += cans[i] * t;  // modification
       }
       for (int i = 1; i <= n; i++) printf("%.3f ", ans[i]);
     }
     ```
 
 * * *
 
-### 例 2 [「BZOJ 3680」吊打 XXX](https://www.luogu.com.cn/problem/P1337) 
+### Sample problem 2 [「BZOJ 3680」Balance point/KO XXX](https://www.luogu.com.cn/problem/P1337)
```
Comment (ksyx) [inline docs/misc/hill-climbing.md:88]: :+1:

File: docs/misc/hill-climbing.md [hunk 10]
```diff
@@ -129,12 +132,12 @@
...
 * * *
 
-## 优化
+## Optimization
 
-很容易想到的是，为了尽可能获取优秀的答案，我们可以多次爬山。方法有修改初始状态/修改降温参数/修改初始温度等，然后开一个全局最优解记录答案。每次爬山结束之后，更新全局最优解。
+It is easy to think that we can climb the hill many times to get the best answers possible. Methods like this include modifying the initial state/modifying the cooling parameter/modifying the initial temperature, etc., and then create a global optimal solution to record the answer. After each hill climbing, the global optimal solution is updated.
 
-这样处理可能会存在的问题是超时，在正式考试时请手造大数据测试调参。
+The problem that may exist in this processing is timeout. Please use handmade large dataset to test parameters during the OI competitions.
```
Comment (ksyx) [inline docs/misc/hill-climbing.md:139]: TLE

handmade 不是原文意思啊，是说 generate by your self 大概这个意思

```suggestion
The problem that may exist in this processing is timeout. Please use handmade large dataset to test parameters during competitive programming competitions.
```

**Comment by ksyx** [comment]
upstream [更新](https://github.com/OI-wiki/OI-wiki/pull/2520)

File: docs/misc/hill-climbing.md (hunk 1 above)
Comment (isdanni) [reply docs/misc/hill-climbing.md:13]: 🉑️ 🌚 

ref是[standford slides](https://web.stanford.edu/group/sisl/k12/optimization/MO-unit2-pdfs/2.1intro.pdf) & [wiki](https://en.wikipedia.org/wiki/Unimodality) & [geeks for geeks](https://www.geeksforgeeks.org/mathematics-unimodal-functions-bimodal-functions/)

> A unimodal function has only one minimum and the rest of the graph goes up from there; or one maximum and the rest of the graph goes down.

> Unimodal Function :
> A function f(x) is said to be unimodal function if for some value m it is monotonically increasing for x≤m and monotonically decreasing for x≥m. For function f(x), maximum value is f(m) and there is no other local maximum.

Comment (ksyx) [reply docs/misc/hill-climbing.md]: > 🉑️ 🌚
> 
> ref是[standford slides](https://web.stanford.edu/group/sisl/k12/optimization/MO-unit2-pdfs/2.1intro.pdf) & [wiki](https://en.wikipedia.org/wiki/Unimodality) & [geeks for geeks](https://www.geeksforgeeks.org/mathematics-unimodal-functions-bimodal-functions/)
> 
> > A unimodal function has only one minimum and the rest of the graph goes up from there; or one maximum and the rest of the graph goes down.
> 
> > Unimodal Function :
> > A function f(x) is said to be unimodal function if for some value m it is monotonically increasing for x≤m and monotonically decreasing for x≥m. For function f(x), maximum value is f(m) and there is no other local maximum.

好的 :+1:

File: docs/misc/hill-climbing.md [hunk 11]
```diff
@@ -30,14 +30,14 @@ About cooling: The cooling parameter is a constant slightly less than $1$ , gene
...
-4. Multiply the recorded cans by the temperature, update the center of the sphere, and go back to step 2.
+2. For the current sphere center, find the average of the Euclidean distance from each known point to the center of the sphere.
+3. Traverse all known points. Record a change value $\textit{cans}$ (record each dimension separately) for the Euclidean distance of each point. If it is greater than the average value, add the difference to the change value, otherwise subtract it. In fact, there is no need to check the size, as long as the absolute value is not considered, the coordinates can be directly calculated. This process can be transformed into a new center of the sphere pushing it back and forth, {pull|push}ing it when it is too {far|close} to the center of sphere.
```
Comment (ksyx) [inline docs/misc/hill-climbing.md:39]: 🌚

Comment (isdanni) [reply docs/misc/hill-climbing.md:39]: 突然反应过来好像不是直接copy&paste 🌚 💦

**Pull request merged**